patches/tModLoader/Terraria/ModLoader/ModAchievement.cs

```diff
@@ -0,0 +1,102 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.Xna.Framework.Graphics;
+using ReLogic.Content;
+using Terraria.Achievements;
+using Terraria.GameContent.Achievements;
+using Terraria.Localization;
+
+namespace Terraria.ModLoader;
+
+/// <summary>
+/// Define a custom achievement and implement how it should act upon completion
+/// </summary>
+public abstract class ModAchievement : ModType<Achievement, ModAchievement>, ILocalizedModType
+{
+    public Achievement Achievement => Entity;
+
+    private string TextureName => (GetType().Namespace + "." + Name).Replace('.', '/');//GetType().FullName.Replace('.', '/');
+    public Asset<Texture2D> Texture { get; private set; }
+    public string LocalizationCategory => "Achievements";
+
+    public override sealed bool IsCloneable => false;
+
+    /// <summary>
+    /// Gets the localized friendly name of the achievement.
+    /// </summary>
+    public virtual LocalizedText FriendlyName => Mod.GetLocalization($"{LocalizationCategory}.{Name}.Name");
+
+    /// <summary>
+    /// Gets the localized description of the achievement.
+    /// </summary>
+    public virtual LocalizedText Description => Mod.GetLocalization($"{LocalizationCategory}.{Name}.Description");
```

Comment:
As per other `ModType` examples, should use `this.GetLocalization(nameof(Description));`